renderdoc/driver/vulkan/vk_core.h

```diff
@@ -755,6 +755,7 @@ class WrappedVulkan : public IFrameCapturer
   VulkanCreationInfo m_CreationInfo;
 
   std::map<ResourceId, std::vector<EventUsage>> m_ResourceUses;
+  std::map<uint32_t, std::set<ResourceUsage>> m_EventResourceUsage;
```

Comment:
This is a fairly heavy way to solve the problem, when at the moment all we care about is "for each EID, did any vertex pipeline shader use RW resources?". We could define each `ResourceUsage` as a bit and fit in a 64-bit bitfield, but that will probably break in future. Instead I think we should define an enum with per-event flags, and one of the bits can be `VertexPipeRWUsage`, and then process it below where we add to `m_EventResourceUsage` to set that bit. This might also be useful for other per-event flags later.

Reply:
I added a commit with modifications to make it easier to see changes (but I will squash before submitting). Let me know if this is what you had in mind.

Reply:
Hmm, I was actually thinking even simpler to just keep this enum local to the vulkan driver. There's no need right now to make the enum part of the public replay interface when nothing will actually use it publicly (maybe not ever). My `VertexPipeRWUsage` was just an example of a flag to cover anything in the vertex pipeline - from vertex to tesellation & geometry stages.

Reply:
Ah that makes a lot of sense to keep that local. Changed, though I think I would want to have a flag per vertex, tess control, tess eval, and geometry shader. They each have a separate shader module, and we have that information already, so we would know which shader stage might need re-writing.